Source/JavaScriptCore/assembler/AbstractMacroAssembler.h

```diff
@@ -1014,21 +1014,23 @@ class AbstractMacroAssembler : public AbstractMacroAssemblerBase {
         return AssemblerType::getCallReturnOffset(call.m_label);
     }
 
-    template<PtrTag jumpTag, PtrTag destTag>
+    template<bool atomic = false, PtrTag jumpTag, PtrTag destTag>
     static void repatchJump(CodeLocationJump<jumpTag> jump, CodeLocationLabel<destTag> destination)
     {
-        AssemblerType::relinkJump(jump.dataLocation(), destination.dataLocation());
+        constexpr auto repatch = atomic ? jitMemcpyAtomicRepatchFlush : jitMemcpyRepatchFlush;
+        AssemblerType::template relinkJump<repatch>(jump.dataLocation(), destination.dataLocation());
```

Comment:
It seems that the real need is only to convey whether this needs to be atomic or not.  Is there really a need to use the `RepatchingInfo` abstraction?  I wonder if it is something that should be constrained to the ARMv7 only implementation.  It's fair to have the high level shared code here pass in that atomicity is required, but it seems that flushing is implied to always be needed here.  I wonder if this is also true elsewhere.

Note: if atomicity is really the only unique piece of info needed, then it makes it clear why ARM64 and X86_64 ports can completely ignore the `atomic` template parameter, because their repatching are already always atomic.  It'd be nice if that's how it plays out.